src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -3278,11 +3278,19 @@ MavlinkReceiver::run()
 			usleep(10000);
 		}
 
-		const hrt_abstime t = hrt_absolute_time();
+		const hrt_abstime now = hrt_absolute_time();
+
+		CheckHeartbeats(now);
 
-		CheckHeartbeats(t);
+		// FIXME:
+		// Mavlink receiver is emitting messages here. That's not what a "receiver" does.
+		// - mission items
+		// - params
+		// - ftp
```

Comment:
I know, I was confused by this as well the other day!

I think the instances should ultimately live in main and then send messages from there and get callbacks when messages are received.

Reply:
Yeah I discovered this a few years ago. And agreed all transmission should occur in the TX thread via queues/callbacks. I'm actually a little confused on how this even works sharing the UART fd between the two threads. I'd also like to rename them to `mavlink_tx_<instance>` and `mavlink_rx_<instance>`